packages/daf-cli/src/server.ts

```diff
@@ -27,150 +27,162 @@ program
-      basePath: apiBasePath,
-      getAgentForRequest: async (req) => agent,
-      exposedMethods,
-      serveSchema: false,
-    })
+    /**
+     * Handling 'did:web' requests ('/.well-known/did.json' and '/^\/(.+)\/did.json$/')
+     */
+    console.log('📋 DID Document ' + baseUrl + didDocEndpoint)
+    app.use(DidDocRouter({ getAgentForRequest }))
+
+    console.log('📖 API Documentation', baseUrl + options.apiDocsPath)
+    app.use(
+      options.apiDocsPath,
+      swaggerUi.serve,
+      swaggerUi.setup(undefined, {
+        swaggerOptions: {
+          url: baseUrl + options.schemaPath,
+        },
+      }),
+    )
 
-    app.use(apiBasePath, passport.authenticate('bearer', { session: false }), agentRouter)
+    /**
+     * Creating server identity and configuring messaging service endpoint
+     */
+    let serverIdentity: IIdentity
+    if (options.defaultIdentity.create) {
+      serverIdentity = await agent.identityManagerGetOrCreateIdentity({
+        provider: 'did:web',
+        alias: hostname,
+      })
+      console.log('🆔', serverIdentity.did)
+
+      const messagingServiceEndpoint = baseUrl + options.defaultIdentity.messagingServiceEndpoint
+
+      console.log('📨 Messaging endpoint', messagingServiceEndpoint)
+      await agent.identityManagerAddService({
+        did: serverIdentity.did,
+        service: {
+          id: serverIdentity.did + '#msg',
+          type: 'Messaging',
+          description: 'Handles incoming POST messages',
+          serviceEndpoint: messagingServiceEndpoint,
+        },
+      })
 
-    app.listen(cmd.port || options.port, async () => {
-      console.log(`🚀 Agent server ready at http://localhost:${cmd.port || options.port}`)
-      console.log('🧩 Available methods', agent.availableMethods().length)
-      console.log('🛠  Exposed methods', exposedMethods.length)
-
-      let baseUrl = options.baseUrl
-
-      if (options.ngrok?.connect) {
-        baseUrl = await ngrok.connect({
-          addr: cmd.port || options.port,
-          subdomain: options.ngrok.subdomain,
-          region: options.ngrok.region,
-          authtoken: options.ngrok.authtoken,
+      app.post(
+        options.defaultIdentity.messagingServiceEndpoint,
+        express.text({ type: '*/*' }),
+        async (req, res) => {
+          try {
+            const message = await agent.handleMessage({ raw: req.body, save: true })
+            console.log('Received message', message.type, message.id)
+            res.json({ id: message.id })
+          } catch (e) {
+            console.log(e)
+            res.send(e.message)
+          }
+        },
+      )
+    }
+
+    /**
+     * Serving homepage
+     */
+    app.get('/', async (req, res) => {
+      const links = [
+        { label: 'API Docs', url: options.apiDocsPath },
+        { label: 'API Schema', url: options.schemaPath },
+        { label: 'DID Document', url: didDocEndpoint },
+      ]
+
+      let verifiablePresentation
```

Comment:
I don't think it's a good idea to list all the presentations without auth.
This is great when getting started, and for debugging but it should be opt-in, at least.

Reply:
Added explanation of how this works. This is experimental

Reply:
👍 saw it.
I still think this should be opt-in, though?